OpenRA.Mods.D2k/Traits/Buildings/ProductionFromMapEdge.cs

```diff
@@ -17,62 +18,100 @@
 namespace OpenRA.Mods.D2k.Traits
 {
 	[Desc("Produce a unit on the closest map edge cell and move into the world.")]
-	class ProductionFromMapEdgeInfo : ProductionInfo
+	class ProductionFromMapEdgeInfo : ProductionInfo, UsesInit<ProductionSpawnLocationInit>
 	{
 		public override object Create(ActorInitializer init) { return new ProductionFromMapEdge(init, this); }
 	}
 
 	class ProductionFromMapEdge : Production
 	{
+		readonly CPos spawnLocation;
+		readonly RallyPoint rp;
+		readonly DomainIndex domainIndex;
+
 		public ProductionFromMapEdge(ActorInitializer init, ProductionInfo info)
-			: base(init, info) { }
+			: base(init, info)
+		{
+			rp = init.Self.TraitOrDefault<RallyPoint>();
```

Comment:
This will introduce a yaml-ordering dependency.  Please `ProductionFromMapEdgeInfo: INotifyCreated` and push this into the `Created()` method.